install-deps.sh

```diff
@@ -425,6 +425,10 @@ else
         echo "Using zypper to install dependencies"
         zypp_install="zypper --gpg-auto-import-keys --non-interactive install --no-recommends"
         $SUDO $zypp_install systemd-rpm-macros
+        if [ -e /usr/bin/python2 ] ; then
+            # see https://tracker.ceph.com/issues/23981
+            $SUDO $zypp_install python2-virtualenv python2-devel || exit 1
```

Comment:
@smithfarm instead of forcing a python3-ready distro to test python2, i am wondering if we can just test python3 on them? i.e., do you think it makes sense to revise #31525 so instead of trying to run "make check" with python3 on all platforms, we can repurpose it to just run python3 tests on **some** distros. yes, this time, i am thinking about making SUSE an exception by passing
```
-DWITH_PYTHON2=OFF -DWITH_PYTHON3=3 -DMGR_PYTHON_VERSION=3
```
to cmake if `$ID` in `opensuse*|suse|sles`. as probably it's the only python3-ready distro at this moment. probably a recent fedora release is also python3-ready... but we don't use it for testing.

what do you think?

Reply:
@tchaikov `do_cmake.sh` already passes the correct py3 cmake options for SUSE:

```
      opensuse*|suse|sles)
          PYBUILD="3"
...
if [[ "$PYBUILD" =~ ^3(\..*)?$ ]] ; then
    ARGS+=" -DWITH_PYTHON2=OFF -DWITH_PYTHON3=${PYBUILD} -DMGR_PYTHON_VERSION=${PYBUILD}"
fi
```

so I don't think you need to add that.

I would be in favor of not testing py2 on SUSE (or all bona fide py3 distros) at all, but isn't #31525 blocked right now?

Reply:
> @tchaikov `do_cmake.sh` already passes the correct py3 cmake options for SUSE:

`do_cmake.sh` is not used by `run-make-check.sh` , which is used by our CI for launching "make check".

#31525 was blocked because i thought i could run "make check" and dashboard tests in a python3 only environment. but it was proved to be too ambitious and not realistic. as dashboard tests rely on python2 packages and their python3 counterparts are not available, even if i compiled python3.6 and boost for xenial to replace ancient python3.5.

so, as i put in last comment. i think we could be more practical by limiting the scope of that PR to only SUSE distros.

Reply:
I am seeing in `run-make-check.sh` the following:

```
source src/script/run-make.sh
```

and

```
configure "-DWITH_GTEST_PARALLEL=ON -DWITH_FIO=ON -DWITH_SEASTAR=ON -DWITH_CEPHFS_SHELL=ON -DWITH_SPDK=ON -DENABLE_GIT_VERSION=OFF $@"
```

and in `src/script/run-make.sh` I am seeing:

```
function configure() {
    local cmake_build_opts=$(detect_ceph_dev_pkgs)
    $DRY_RUN ./do_cmake.sh $cmake_build_opts $@ || return 1
}
```

What am I missing here?

Reply:
I totally agree to stop testing python2 ASAP on any distros where it is possible. But I see that effort as orthogonal to this PR.

Reply:
no, it's related.

per your commit message

> tox will try to run unit tests ("make check" tests) on both. But the spec-file-based mechanism used by install-deps.sh for determining "make check" dependencies only

if we don't prepare the wheels for python2, and don't test with python2. the problem goes away.